### PR TITLE
redirect legacy search pages to canonical

### LINKED ISSUE #6: Redirect legacy search pages to canonical
Currently the search page format is as follows:

`golang.cafe/Golang-{tag}-Jobs-In-{location}`

Where `tag` can either be a skill (DevOps, SysAdmin, Backend) or a company name (Google, Apple)

There are some legacy pages which used to use query string instead of this canonical format. Here's the legacy format

`golang.cafe/?l={location}&t={tag}`

The above should emit a 301 translating the request URI into the new canonical format.

Eg. `golang.cafe/?l=London&t=DevOps` should redirect to `golang.cafe/Golang-DevOps-Jobs-In-London`

Current handler is here https://github.com/golang-cafe/golang.cafe/blob/3c02ac4/pkg/handler/std_handler.go#L19-L27

### PR DESCRIPTION
**Pull request by duartesantos**
fixes #6

### CODE REVIEW

File: pkg/handler/std_handler.go [hunk 1]
```diff
@@ -21,8 +21,11 @@ func IndexPageHandler(svr server.Server) http.HandlerFunc {
 		location := r.URL.Query().Get("l")
 		tag := r.URL.Query().Get("t")
 		page := r.URL.Query().Get("p")
-		// @TODO: redirect to canonical
-		svr.RenderPageForLocationAndTag(w, location, tag, page, "landing.html")
+		dst := fmt.Sprintf("/Golang-%s-Jobs-In-%s", tag, location)
```
Comment (0x13a) [inline pkg/handler/std_handler.go]: it's a bit more complex than that. If tag or location are empty then the format is going to be different.

For eg.

scenario 1
tag = "DevOps"
location = "NYC"
dst = /Golang-DevOps-Jobs-In-NYC

scenario 2
tag = ""
location = "NYC"
dst = /Golang-Jobs-In-NYC

scenario 3
tag = "DevOps"
location = ""
dst = /Golang-DevOps-Jobs

scenario 4
tag = ""
location = ""
dst = /

Comment (duartesantos) [reply pkg/handler/std_handler.go]: completely missed those variants 🤦‍♂

File: pkg/handler/std_handler.go [hunk 2]
```diff
@@ -21,8 +21,23 @@ func IndexPageHandler(svr server.Server) http.HandlerFunc {
...
+			dst = fmt.Sprintf("/Golang-%s-Jobs", tag)
+		}
+		if dst != "" && page != "" {
+			dst += fmt.Sprintf("?p=%s", page)
+		}
+		if dst != "" {
+			svr.Redirect(w, r, http.StatusMovedPermanently, dst)
+		}
+
+		svr.RenderPageForLocationAndTag(w, "", "", page, "landing.html")
```
Comment (0x13a) [inline pkg/handler/std_handler.go]: this should be removed, default redirect destination should be "/"

**Comment by 0x13a** [comment]
great thanks @duartesantos!

**Pull request merged**